### ISSUE TITLE
Websocket doesn't stay open?

### ISSUE THREAD

**Issue by stevepryde** [Synphonyte/leptos-use#115]
I'm trying to use `use_websocket_with_options()` but after about a minute the websocket closes and doesn't reconnect even though I've set the reconnect options on it.

```rust
let UseWebsocketReturn { message, send, .. } = use_websocket_with_options(
    &ws_url,
    UseWebSocketOptions::default()
        .immediate(true)
        .reconnect_limit(u64::MAX)
        .reconnect_interval(1000)
);
```

I noticed that the websocket demos at https://leptos-use.rs/network/use_websocket.html also do not stay open. They close after about the same amount of time as I've observed locally.

It's quite likely I am doing something wrong. Currently I'm storing the following in context:
```rust
pub struct WebsocketContext {
    pub message: Signal<Option<String>>,
    pub sender: Rc<dyn Fn(String)>,
}
```

I have two main questions:
1. I didn't notice any heartbeat messages getting sent (usually websockets send a heartbeat every 25 seconds). Do I need to do this manually?
2. How can I get the websocket to automatically reconnect?

**Comment by maccesch** [comment]
Interesting observation! Thanks for reporting this. 

Indeed we don't make any assumptions about how you want to use the websocket. So you will need to send a heartbeat manually. I'll be adding an enhancement issue to support this directly in leptos-use as it will be useful for many. Do you think we should make this an opt-in or opt-out feature?

The reconnecting should work already. I'll have to check that.

**Comment by stevepryde** [comment]
Actually I wasn't sure whether the server or the client is supposed to send the heartbeat. For now I just made my server send one and that will keep the websocket open.

It still doesn't reconnect on its own. I wired up a signal to the on_close hook and I have it reconnecting now. It seemed from the docs that it should do that automatically but for whatever reason it doesn't reconnect by default for me.
